### ISSUE TITLE
Building luv as a shared library

### ISSUE THREAD

**Issue by joerg-krause** [luvit/luv#138]
To link against luv it is necessary to build luv as a shared library. Currently, luv is build as a `MODULE` library that is not intenteded to be linked into other targets but may be loaded dynamically at runtime.

Is there any reason to build luv **not** as a shared library?

**Comment by creationix** [comment]
My understanding was that `MODULE` was more appropriate (especially on OSX) since that's how lua loads native modules.  For projects where I want to include luv, I don't use the cmake file in luv.  Luvi, for example duplicates bits of the cmake in it's own.

**Comment by joerg-krause** [comment]
A `MODULE` is a kind of plugin which may be loaded dynamically at runtime using dlopen-like functionality. [CMake Documentation] (http://www.cmake.org/cmake/help/v3.2/command/add_library.html).

Lua and LuaJIT (via `ffi.load`) need shared (dynamic) libraries for loading native libaries.

As far as I understand both, `SHARED` library and `MODULE`, are dynamic libraries with the distinction that you cannot link to a `MODULE` at compile-time.

I'd like to link luvi to luv in Buildroot. I already made some (non-puplished) changes to luv and luvi to get it work. There are some problems which need to be solved:
1. CMake does not install luv as a shared library with its type being set to `MODULE`
2. To use luv as shared library on Unix system it has to follow the `soname` naming convention. This means it has to be prefixed with `lib` and a version number has to appended after the `so` suffix.
3. Some header files must be installed to the target system.

I found a discussion on [stackoverflow](http://stackoverflow.com/questions/4845984/difference-between-modules-and-shared-libraries) about the difference of `SHARED` and `MODULE` libraries.

In my opinion it should be save to build luv as shared library with the benefit that external applications can link against luv.

I will send a PR for testing.
